warp10/src/main/java/io/warp10/script/functions/FOREACH.java

```diff
@@ -69,12 +69,29 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
       throw new WarpScriptException(getName() + " expects a macro on top of the stack.");
     }
     
-    if (!(obj instanceof List) && !(obj instanceof Map) && !(obj instanceof Iterator) && !(obj instanceof Iterable) && !(obj instanceof GeoTimeSerie) && !(obj instanceof GTSEncoder)) {
-      throw new WarpScriptException(getName() + " operates on a list, map, Geo Time Series™, ENCODER, iterator or iterable.");
+    if (!(obj instanceof List) && !(obj instanceof Map) && !(obj instanceof Iterator) && !(obj instanceof Iterable) && !(obj instanceof GeoTimeSerie) && !(obj instanceof GTSEncoder) && !(obj instanceof String)) {
+      throw new WarpScriptException(getName() + " operates on a list, map, Geo Time Series™, ENCODER, STRING, iterator or iterable.");
     }
 
     long index = 0;
-
+    
+    if (obj instanceof String) {
+      final String s = (String) obj;
+      
+      obj = new Iterator<String>() {        
+        int idx = 0;
```

Comment:
Default method 'remove' is not overridden. It would cause compilation problems with JDK 7.

Do we still care about jdk 7 ?

Reply:
JDK7 will be unsupported in future releases. As of 2.4.0 I am not sure everything actually works in JDK7 anyway, and I really doubt any deployment of Warp 10 actually uses JDK7 as its JVM so I think this can be left as is.